File: amss/lib/models/report.dart
import 'user.dart';

class Report {
  int id = -1;
  User user = User();
  String reason = "";

  static Report fromMap(Map<String, dynamic> json) {
    final report = Report();
    report.id = json["id"];
    report.user = User.fromMap(json["user"]);
    report.reason = json["reason"];
    return report;
  }
}

File: backend/src/main/java/com/example/backend/models/Report.java
package com.example.backend.models;

import jakarta.persistence.*;
import jakarta.validation.constraints.NotBlank;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.NoArgsConstructor;

@Builder
@NoArgsConstructor
@AllArgsConstructor
@Entity(name = "reports")
public class Report {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    public Long id;

    @NotBlank
    public String reason;

    @ManyToOne
    public User user;
}
